Sources/AuthFoundation/JWT/Internal/DefaultIDTokenValidator.swift

```diff
@@ -19,56 +19,180 @@ import CommonCrypto
 struct DefaultIDTokenValidator: IDTokenValidator {
     var issuedAtGraceInterval: TimeInterval = 300
     
+    ///  This represents the validation checks using to  validate the claim
+    enum Intent {
```

Comment:
Supplying the token, context, issuer, etc here seems to complicate the enum without adding much value, because these arguments are passed to private functions within the DefaultIDTokenValidator in the end. 

It also runs the risk of unrelated JWT tokens being passed to the same validator call. 

Perhaps a better approach would be to keep the original validator method signature, and add a `tolenValidation` property (or something along those lines) to the Context object. The validation can take those settings and can use that internally to perform validation steps.

You could also supply a static property on the Intent enum called `all`, which not only includes the full set of validation steps, but is the default if none is supplied. 

This is because the developer doesn't directly invoke the validate function, but is rather automatically called after a token is exchanged. So the developer would need to pass these options at the time they initiate an authentication flow so as to control the behavior later on downstream. 

An alternative might be to make this a property on the token validation itself, that can be assigned at app launch. But that then means different tokens can't be validated with different verification steps (which I could imagine would be important to some use-cases).

Reply:
Thanks alot @mikenachbaur-okta . There's alot of good callout there. 
- I'll make changes to the Intent, we definitely do not need to supply the token, context etc to the intent. 
- I also think we can use an OptionSet here as well to define the order.

Reply:
I've updated this.